addons/urdfreader/urdfreader.cc

```diff
@@ -280,9 +275,10 @@ void construct_model(Model *rbdl_model, ModelPtr urdf_model,
         I_child->origin.position.x,
         I_child->origin.position.y,
         I_child->origin.position.z);
-      I_child->origin.rotation.RPY(link_inertial_rpy[0],
-                                   link_inertial_rpy[1],
-                                   link_inertial_rpy[2]);
+      I_child->origin.rotation.RPY(link_inertial_rpy_temp.x,
+                                   link_inertial_rpy_temp.y,
+                                   link_inertial_rpy_temp.z);
+      link_inertial_rpy.set(link_inertial_rpy_temp.x, link_inertial_rpy_temp.y, link_inertial_rpy_temp.z);
```

Comment:
same here? Why the temporary variable?

Reply:
The reason the `_temp` variables were created for this and the above snippets is because the `urdf_joint->PARENT_TRANSFORM.rotation.RPY` command requires floating point (double) references as arguments. This will not compile if you try to use `joint_rpy` in the Casadi version because `Vector3d` in that case is a vector of symbolics (MX). I couldn't think of a cleaner way to do this, but if you have ideas let me know.

Reply:
okay, maybe add a small comment to make this clear ;)

Reply:
Added https://github.com/PeterQLee/rbdl/blob/master/addons/urdfreader/urdfreader.cc#L240-L241